python/cudf/cudf/core/frame.py

```diff
@@ -3581,64 +3582,148 @@ def _reindex(
         return self._mimic_inplace(result, inplace=inplace)
 
 
-def _get_replacement_values(to_replace, replacement, col_name, column):
+def _get_replacement_values_for_columns(
+    to_replace: Any, value: Any, columns: Dict[Any, Any]
+) -> Tuple[Dict[Any, Any], Dict[Any, Any], Dict[Any, Any]]:
+    """
+    Returns a per column mapping for the values to be replaced, new
+    values to be replaced with and if all the values are empty.
 
-    all_nan = False
-    if (
-        is_dict_like(to_replace)
-        and not isinstance(to_replace, cudf.Series)
-        and replacement is None
+    Parameters
+    ----------
+    to_replace : numeric, str, list-like or dict
+        Contains the values to be replaced.
+    value : numeric, str, list-like, or dict
+        Contains the values to replace `to_replace` with.
+    columns : dict
+        A column to dtype mapping representing dtype of columns.
```

Comment:
Could we rename this parameter? If this is a mapping of columns -> dtypes then calling it `columns` isn't intuitive

Reply:
Renamed this parameter